src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -29,24 +31,35 @@
 @RequestMapping("/decoder")
 @RequiredArgsConstructor
 public class DecoderController {
+
   public static final String UNKNOWN_ERROR_MESSAGE = "Unknown error in DecoderController";
 
   private final DecoderDaoImpl decoderDao;
   private final DeviceDaoImpl deviceService;
   private final DecoderMapper decoderMapper;
   private final StreamMapper streamMapper;
   private final DecoderService decoderService;
+  private final MaintainDeviceStatus maintainDeviceStatus;
 
   @GetMapping
   public List<DecoderDto> retrieveAllDecoders() {
     List<DecoderEntity> decoderEntity = decoderDao.getDecoders();
+    maintainDeviceStatus.maintainStatusField(decoderEntity);
     return decoderMapper.toDecoderDtos(decoderEntity);
   }
 
   @GetMapping("/{serialNumber}")
   public ResponseEntity<DecoderDto> retrieveDecoder(@PathVariable @Valid String serialNumber) {
-    return decoderDao
-        .findDecoder(serialNumber)
+    // maintain status field and create a log if status changed
+    Optional<DecoderDto> decoder = decoderDao.findDecoder(serialNumber);
+    if (decoder.isPresent()) {
+      List<DecoderEntity> decodersListTemp = new LinkedList<>();
+      decodersListTemp.add(decoderMapper.toDecoderEntity(decoder.get()));
+      maintainDeviceStatus.maintainStatusField(decodersListTemp);
+    }
+
+    // return decoder
+    return decoder
```

Comment:
i think its better to put all of the logging related stuff .
i.e.
- maintain the status 
- return the properly maintained status
- use the aspect to do the logging part "afterReturning"

Reply:
maybe even maintaing the status could be done in service level and not util  (i.e. retrieveDecoder in service). what do you think. if itll be used by other functions then keep it at util if not then at service maybe better

Reply:
Good suggestion! I have extracted the the logging logic out of the MaintainDeviceStatus class to an Aspect class. For maintaining the status in the service level, I decided not to proceed for two reasons: 1- It's going to require more work on this US while we have others to tackle 2- I don't think there's a huge benefit from doing such a thing.